### PR TITLE
Update typescript setup

### PR DESCRIPTION
**Pull request by benmccann**
- **Quick Checklist**

* [X] I have read [the contributing guidelines](../CONTRIBUTING.md)
* [ ] I have written new tests, as applicable (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] I have added a changeset, if applicable

- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Cleanup

- **What is the new behavior (if this is a feature change)?**

None

- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this
  PR?)

No

- **Other information**:

`rollup-plugin-typescript2` has tons of required peerDependencies that aren't installed. Switch to `@rollup/plugin-typescript` instead since it has fewer required peerDependencies and then install them (i.e. `tslib`). `rollup-plugin-typescript2` was originally better than `@rollup/plugin-typescript`, but the latter has improved quite a bit and now is the better library and far better tested (`rollup-plugin-typescript2` has no tests whatsoever)

This is necessary if we'd want to try out pnpm (https://github.com/JonasKruckenberg/imagetools/issues/292) because pnpm validates that you have the required peedDependencies installed

The `yarn.lock` file is smaller with this change because `@rollup/plugin-typescript` pulls in fewer dependencies than `rollup-plugin-typescript2` does, which is nice as well

### CODE REVIEW

**Comment by JonasKruckenberg** [comment]
Does `@eollup/plugin-typescript` generate typings now? That was the only reason for `rollup-plugin-typescript2`.

**Comment by benmccann** [comment]
Yes, I believe it should: https://github.com/rollup/plugins/issues/61

I'm not sure if there might be some additional configuration needed for that to happen or if it's already happening with this PR

**Comment by benmccann** [comment]
I've pushed a change to make it so that types are generated with this PR

File: packages/core/package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": "3.0.1",
   "main": "dist/index.cjs",
   "module": "dist/index.mjs",
-  "types": "dist/index.d.ts",
+  "types": "dist/types.d.ts",
```
Comment (JonasKruckenberg) [inline packages/core/package.json:6]:
```suggestion
  "types": "dist/index.d.ts",
```

`index.d.ts` exports all type declarations (including functions and objects), `typed.d.ts` is just exporting common interface declarations and type aliases.

File: packages/rollup/package.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "version": "3.0.3",
   "main": "dist/index.cjs",
   "module": "dist/index.mjs",
-  "types": "dist/rollup/src/index.d.ts",
+  "types": "dist/types.d.ts",
```
Comment (JonasKruckenberg) [inline packages/rollup/package.json:6]:
```suggestion
  "types": "dist/index.d.ts",
```

File: packages/vite/package.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
   "version": "4.0.2",
   "main": "dist/index.cjs",
   "module": "dist/index.mjs",
-  "types": "dist/vite/src/index.d.ts",
+  "types": "dist/types.d.ts",
```
Comment (JonasKruckenberg) [inline packages/vite/package.json:7]:
```suggestion
  "types": "dist/index.d.ts",
```

**Review by JonasKruckenberg — CHANGES_REQUESTED** [review]

**Comment by benmccann** [comment]
this is strange. I can't reproduce any failure locally eventhough the CI keeps failing

**Comment by JonasKruckenberg** [comment]
We love jest 🙃

**Comment by benmccann** [comment]
Maybe I'll finally have a reason to try out the new https://vitest.dev/

**Comment by JonasKruckenberg** [comment]
I ended up implementing this myself, to fix that annoying jest issue, but thanks for your PR anyway!
repo uses pnpm now 😉

**Comment by benmccann** [comment]
that's great! thank you so much!

**Pull request closed without merging**